consul/client.go

```diff
@@ -182,10 +182,41 @@ func isKeyIndexMismatchErr(err error) bool {
 func newKvClient(machine string, p connectionParams) (*api.Config, *api.Client, error) {
 	logrus.Infof("consul: connecting to %v", machine)
 	config := api.DefaultConfig()
-	config.HttpClient = http.DefaultClient
-	config.Address = machine
-	config.Scheme = "http"
-	config.Token = p.options[kvdb.ACLTokenKey]
+
+	// check if TLS is required
+	caFile, ok1 := p.options[kvdb.CAFileKey]
+	certFile, ok2 := p.options[kvdb.CertFileKey]
+	certKeyFile, ok3 := p.options[kvdb.CertKeyFileKey]
+	caAuthAddress := p.options[kvdb.CAAuthAddress]
+	insecureSkipVerify := strings.ToLower(p.options[kvdb.InsecureSkipVerify]) == "true"
+	if ok1 && ok2 && ok3 {
+		tlsConfig := &api.TLSConfig{
+			CAFile:             caFile,
+			CertFile:           certFile,
+			KeyFile:            certKeyFile,
+			Address:            caAuthAddress,
+			InsecureSkipVerify: insecureSkipVerify,
+		}
+
+		consulTLSConfig, err := api.SetupTLSConfig(tlsConfig)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		config := api.DefaultConfig()
+		config.Address = machine
+		config.Scheme = "https"
+		config.HttpClient = new(http.Client)
+
+		config.HttpClient.Transport = &http.Transport{
+			TLSClientConfig: consulTLSConfig,
+		}
```

Comment:
Do we need to set the ACL token key here?

Reply:
i moved the code around to take ACL token input.